client/src/modules/dashboard/Events/components/EventForm.tsx

```diff
@@ -108,6 +136,72 @@ const EventForm: React.FC<EventFormProps> = (props) => {
           </FormControl>
         )}
 
+        <FormControl id="first-name" isRequired>
+          <Box display="flex" alignItems="end" m="1">
+            <FormLabel> Sponsers</FormLabel>
+            <Spacer />
+            <Button
+              onClick={() => {
+                append({
+                  type: 'FOOD',
+                  id: 0,
+                });
+              }}
+            >
+              Add
+            </Button>
+          </Box>
+          {sponsors.map((sponsor, index) => {
+            return (
```

Comment:
```suggestion
            const registeredSponsor = register(
              `sponsors.${index}.type` as const,
              {
                required: true,
              },
            );
            return (
```
Used below for a custom `onChange`